### PR TITLE
[SPARK-43687][SPARK-43688][SPARK-43689][SPARK-43690][PS] Fix `NumOps` for Spark Connect

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to fix `NullOps` test for pandas API on Spark with Spark Connect.

This includes SPARK-43687, SPARK-43688, SPARK-43689, SPARK-43690 at once, because they are all related similar modifications in single file.

### Why are the changes needed?

To support all features for pandas API on Spark with Spark Connect.

### Does this PR introduce _any_ user-facing change?

Yes, `NumOps.lt`,  `NumOps.le`, `NumOps.ge`, `NumOps.gt` are now working as expected on Spark Connect.

### How was this patch tested?

Uncomment the UTs, and tested manually.

### CODE REVIEW

File: python/pyspark/pandas/data_type_ops/num_ops.py [hunk 1]
```diff
@@ -215,43 +215,35 @@ def abs(self, operand: IndexOpsLike) -> IndexOpsLike:
...
-        if is_remote():
-            from pyspark.sql.connect.column import Column as ConnectColumn
-
-            Column = ConnectColumn
-        else:
-            Column = PySparkColumn  # type: ignore[assignment]
-        return column_op(Column.__lt__)(left, right)  # type: ignore[arg-type]
+        result = pyspark_column_op("__lt__")(left, right)
+        if is_remote:
+            # In Spark Connect, it returns None instead of False, so we manually cast it.
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/data_type_ops/num_ops.py:220]: Can you file a JIRA? cc @ueshin

File: python/pyspark/pandas/data_type_ops/num_ops.py [hunk 2]
```diff
@@ -215,43 +215,35 @@ def abs(self, operand: IndexOpsLike) -> IndexOpsLike:
...
         _sanitize_list_like(right)
-        if is_remote():
-            from pyspark.sql.connect.column import Column as ConnectColumn
-
-            Column = ConnectColumn
-        else:
-            Column = PySparkColumn  # type: ignore[assignment]
-        return column_op(Column.__lt__)(left, right)  # type: ignore[arg-type]
+        result = pyspark_column_op("__lt__")(left, right)
+        if is_remote:
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/data_type_ops/num_ops.py:219]: I believe this is a function

File: python/pyspark/pandas/data_type_ops/num_ops.py (hunk 1 above)
Comment (itholic) [reply python/pyspark/pandas/data_type_ops/num_ops.py]: Just created a JIRA (SPARK-43877) & updated the comment.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**